src/telemetry/telemetry_metadata.c

```diff
@@ -15,6 +17,72 @@
 #include "scan_iterator.h"
 #include "jsonb_utils.h"
 
+#if PG14_LT
+/* Copied from jsonb_util.c */
+static void
+JsonbToJsonbValue(Jsonb *jsonb, JsonbValue *val)
+{
+	val->type = jbvBinary;
+	val->val.binary.data = &jsonb->root;
+	val->val.binary.len = VARSIZE(jsonb) - VARHDRSZ;
+}
+#endif
+
+void
+ts_telemetry_event_truncate(void)
+{
+	RangeVar rv = {
+		.schemaname = CATALOG_SCHEMA_NAME,
+		.relname = TELEMETRY_EVENT_TABLE_NAME,
+	};
+	ExecuteTruncate(&(TruncateStmt){
+		.type = T_TruncateStmt,
+		.relations = list_make1(&rv),
+		.behavior = DROP_RESTRICT,
+	});
+}
+
+void
+ts_telemetry_events_add(JsonbParseState *state)
+{
+	ScanIterator iterator =
+		ts_scan_iterator_create(TELEMETRY_EVENT, AccessShareLock, CurrentMemoryContext);
```

Comment:
there doesn't seem to be any limit on the amount of these events being added - shouldn't there be one?

I wonder how realistic of the case of having a request pumped up by many events; failing to be processed on the other end - so that it will just keep failing again and again...

Reply:
I'm just saying that it doesn't have any bounds - which might cause it to hit back in possibly odd situations....

let me rephrase it differently: if for some reason the target service where this is connecting can't be reached (say firewall rules) - the telemetry events will keep accumlating ; and the system will build a bigger and bigger request (for no good reason).

I don't know how frequently events will be produced and at what rate...
if you think its not worth the effort - it could be left as-is.

Reply:
> I don't know how frequently events will be produced and at what rate...
if you think its not worth the effort - it could be left as-is.

It is not yet used by anything and we do not expect this to generate a lot of data.